src/SDPoptimize.jl

```diff
@@ -238,168 +142,170 @@ function sdp_optimize(model::SPModel,
                   display=true::Bool)
 
     TF = model.stageNumber
-    V = zeros(Float64, param.totalStateSpaceSize, TF+1)
-    x = zeros(Float64, model.dimStates)
-    x1 = zeros(Float64, model.dimStates)
+    next_state = zeros(Float64, model.dimStates)
     law = model.noises
 
-    u_lower_bounds = [ i for (i , j) in model.ulim]
-    x_lower_bounds = [ i for (i , j) in model.xlim]
+    u_bounds = model.ulim
+    x_bounds = model.xlim
+    x_steps = param.stateSteps
 
-    count_iteration = 1
+    #Compute cartesian product spaces
+    product_states, product_controls = generate_grid(model, param)
+
+    V = zeros(Float64, param.stateVariablesSizes..., TF)
 
     #Compute final value functions
-    for indx = 1:(param.totalStateSpaceSize)
-        x = variable_from_index(indx, x_lower_bounds, param.stateVariablesSizes,
-                                param.stateSteps)
-        V[indx, TF+1] = model.finalCostFunction(x)
+    for x in product_states
+        ind_x = index_from_variable(x, x_bounds, x_steps)
+        V[ind_x..., TF] = model.finalCostFunction(x)
     end
 
     #Construct a progress meter
-    p = Progress(TF*param.totalStateSpaceSize, 1)
+    if display
+        p = Progress((TF-1)*param.totalStateSpaceSize, 1)
+    end
 
     #Display start of the algorithm in DH and HD cases
-    #Define the loops order in sdp
     if (param.infoStructure == "DH")
```

Comment:
I think we should split DH and HD for the moment, just to have smaller functions. What do you think?